src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -201,6 +219,19 @@ fn check_aliasability<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
     }
 }
 
+fn check_yields<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
```

Comment:
Well, I forgot that `check_loans` uses the `ExprUseVisitor`. I would add a "callback" to that interface for "yields". So, specifically the `expr_use_visitor::Delegate` interface. We could add a `fn yield(id: ast::NodeId)` method. Then, on each callback, we could check whether there are any loans in scope at that point, e.g. by calling `CheckLoans::each_in_scope_loan()`.